pppd/sys-linux.c

```diff
@@ -127,9 +128,27 @@
 #include <linux/ppp_defs.h>
 #include <linux/if_ppp.h>
 
-#ifdef INET6
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <linux/if_link.h>
+/* Attempt at retaining compile-support with older than 4.7 kernels,
+ * older than 2.6.35 will probably break.  Either way, pre 4.7 kernels
+ * will not work with this code. */
```

Comment:
Just to note that this ifdef is compile time dependency, not runtime dependency. It is possible to use reduced header files or SDK (probably in embedded or cross compile toolchains) where not all constants are defined. So this can be required also when using new post 4.7 kernels. IIRC people already reported issues that pppd compilation failed just because they did not have `NETLINK_CAP_ACK` or `SOL_NETLINK` in header files.

Reply:
That comment is broken either way, since now 2.6.35 and older should compile too, so comment updated.